service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/GetGpcStructuredTaskExecutor.java

```diff
@@ -130,4 +159,42 @@ private GetGpcDocumentTaskDefinition buildGetDocumentTask(TaskDefinition taskDef
             .messageId(externalAttachment.getMessageId())
             .build();
     }
+
+    private boolean isLargeEhrExtract(String ehrExtract) {
+        return getBytesLengthOfString(ehrExtract) > gp2gpConfiguration.getLargeEhrExtractThreshold();
+    }
+
+    private OutboundMessage.ExternalAttachment buildExternalAttachment(String chunk, TaskDefinition taskDefinition) {
+        var messageId = randomIdGeneratorService.createNewId();
+        var documentId = randomIdGeneratorService.createNewId();
+        var documentName = GpcFilenameUtils.generateDocumentFilename(
+            taskDefinition.getConversationId(), documentId
+        );
+
+        return OutboundMessage.ExternalAttachment.builder()
+            .documentId(documentId)
+            .messageId(messageId)
+            .contentType("application/xml") // confirm this is correct
+            .filename(documentName)
+            .length(getBytesLengthOfString(chunk))
+            .compressed(true)
+            .largeAttachment(true)
+            .originalBase64(true) // always true since GPC gives us a Binary resource which is mandated to have base64 encoded data
```

Comment:
I'm not quite sure about that. The original was a FHIR Bundle - the Structured Record. The translated data is XML - still not base 64.
Can you check documentation, shouldn't we do base64 encoding before splitting into parts? Otherwise I'm not sure this should be set to true